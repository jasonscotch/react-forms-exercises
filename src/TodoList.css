.TodoList {
    margin: 0 auto;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }
  
  .TodoList-div {
    margin-top: 50px;
    margin: 25px;
    font-weight: bold;
    justify-content: center;
  }
  
  form label {
      margin: 10px;
  }
  form input {
      margin: 5px;
  }
  form button {
      margin: 5px;
  }