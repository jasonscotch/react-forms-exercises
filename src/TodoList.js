import React, { useState } from "react";
import {v4 as uuid} from 'uuid';
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";
import "./TodoList.css";


const TodoList = () => {
    const INITIAL_STATE = [
        {id: uuid(), todo: 'Example Todo'}
    ]
    const [todos, setTodos] = useState(INITIAL_STATE);
    const addTodo = (newTodo) => {
        setTodos(todos => [...todos, { ...newTodo, id: uuid() }])
    }
    const removeTodo = (idRemove) => {
        setTodos(todos => todos.filter(todo => todo.id !== idRemove));
    } 
    const toggleCompleted = (idToToggle) => {
        setTodos((todos) =>
            todos.map((todo) =>
                todo.id === idToToggle
                    ? { ...todo, completed: !todo.completed }
                    : todo
            )
        );
    }

    return (
        <div className="TodoList">
            <h3>ALL THE TODOS!</h3>
            <NewTodoForm addTodo={addTodo} />
            <div className="TodoList-div">
                {todos.map(({ id, todo, completed }) => <Todo id={id} todo={todo} key={id} removeTodo={removeTodo} completed={completed} toggleCompleted={toggleCompleted} />)}
            </div>
        </div>
    )
}

export default TodoList;