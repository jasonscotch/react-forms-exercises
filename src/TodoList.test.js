import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

it("renders without crashing", function () {
  render(<TodoList />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("should add a new box", () => {
    const { queryByText, getByLabelText } = render(<TodoList />);
    const input = getByLabelText("Todo:");
    const btn = queryByText("Add Todo");
    expect(queryByText('Walk Dog')).not.toBeInTheDocument();
    
    fireEvent.change(input, { target: { value: 'Walk Dog' } });
    fireEvent.click(btn);

    expect(queryByText('Walk Dog')).toBeInTheDocument();
});
  
it("should remove a box when X is clicked", () => {
    const { queryByText, getByLabelText, getByText } = render(<TodoList />);
    const input = getByLabelText("Todo:");
    const btn = queryByText("Add Todo");
    
    fireEvent.change(input, { target: { value: 'Walk Dog' } });
    fireEvent.click(btn);

    const xBtn = getByText('X', { selector: '.Todo-container:last-child .X' });

    fireEvent.click(xBtn);
  
    expect(queryByText('Walk Dog')).not.toBeInTheDocument();
    
})
 