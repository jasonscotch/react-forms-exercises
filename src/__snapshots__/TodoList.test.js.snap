// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <h3>
      ALL THE TODOS!
    </h3>
    <form>
      <label
        for="todo"
      >
        Todo:
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="Todo"
        type="text"
        value=""
      />
      <button>
        Add Todo
      </button>
    </form>
    <div
      class="TodoList-div"
    >
      <div
        class="Todo-container"
      >
        <div
          class="X"
        >
          X
        </div>
        <div
          class="Todo "
        >
          Example Todo
        </div>
        <button>
          Mark as Completed
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
